### PR TITLE
Add kubectl overview page

### LINKED ISSUE #11814: Create a kubectl overview page for the user guide
Something like:
https://docs.openshift.org/latest/cli_reference/basic_cli_operations.html

cc @jlowdermilk @deads2k @caesarxuchao

### PR DESCRIPTION
**Pull request by hw-qiaolei**
This is an effort try to fix: #11814

@bgrant0607 @hurf

### CODE REVIEW

**Comment by bgrant0607** [comment]
Thanks!

cc @kubernetes/kubectl

**Comment by bgrant0607** [comment]
@RichieEscarez Please suggest where links to this should be added, or existing links to the generated docs should be changed to point here.

File: docs/user-guide/kubectl-overview.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+- `NAME` is the name of resource(s). `TYPE NAME` can be specified as `TYPE name1 name2` or `TYPE/name1 TYPE/name2`. `TYPE NAME` can also be specified by one or more file arguments: `-f file1 -f file2 ...`, [use YAML rather than JSON](config-best-practices.md) since YAML tends to be more user-friendly for config.
+- `flags` are used to provide more control information when running a command. For example, you can use `-s` or `--server` to specify the address and port of the Kubernetes API server. Command line flags override their corresponding default values and environment variables. [Use short flags sparingly, only for the most frequently used options](../devel/kubectl-conventions.md).
+
+Please use `kubectl help [command]` for detailed information about a command.
+
+Please refer to [kubectl](kubectl/kubectl.md) for a complete list of available commands and flags.
+
+## Common Operations
+
+For explanation, here I gave some mostly often used `kubectl` command examples. Please replace sample names with actually values if you would like to try these commands.
```
Comment (jlowdermilk) [inline docs/user-guide/kubectl-overview.md]: s/actually/actual

**Comment by jlowdermilk** [comment]
This is a good intro page. My only concern is that it will not stay up to date unless it's autogenerated. A good solution would be to modify [gen_kubectl_docs.go](https://github.com/kubernetes/kubernetes/blob/master/cmd/gendocs/gen_kubectl_docs.go) to generate this doc, if you're interested in doing that. In particular the examples, operations, and resource types should be retrieved from the built kubectl. I'm also fine with submitting this as is (once appropriate links to/from existing kubectl docs are updated) and putting a note in #14177 to autogenerate this file.

**Comment by JanetKuo** [comment]
Agree on auto-generate this doc. If we're submitting this as is please add `kubectl edit` too. https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/kubectl/kubectl_edit.md

**Comment by hw-qiaolei** [comment]
@jlowdermilk thank you for reviewing this and giving these good suggestions. Yes this page is manually written and will be out of date if there is a change of `kubectl`. I will look into how to auto-generate it.
My initial thought is that this page should include some most often used examples(not all of the commands), and for readability I rewrote these examples(not all of the examples given by a certain command, and changed some variables, explanations etc). They should be **small and stable, more user-friendly**.
Anyway, I will look into how the auto-generate it and try to keep a balance between **keeping it up with `kubectl` and being small and user-friendly**.

**Comment by hw-qiaolei** [comment]
@JanetKuo thank you for the comment. I have added `kubectl edit` manually and will look into how to auto-generate this doc.

**Comment by hw-qiaolei** [comment]
cc @alfred-huangjian

**Comment by RichieEscarez** [comment]
Sorry for the delay. Starting my review now.

**Comment by RichieEscarez** [comment]
After a discussion with @jlowdermilk, im going to LGTM this to make it live and then open a separate PR for my edits.

**Pull request merged**